packages/libs/error-stack/src/fmt/charset.rs

```diff
@@ -0,0 +1,126 @@
+use core::sync::atomic::{AtomicU8, Ordering};
+
+use crate::{
+    fmt::r#override::{AtomicOverride, AtomicPreference},
+    Report,
+};
+
+/// The available supported charsets
+///
+/// Can be accessed through [`crate::fmt::HookContext::charset`], and set via
+/// [`Report::set_charset`].
+#[derive(Debug, Copy, Clone)]
```

Comment:
We should derive `PartialEq`, `Eq`, and `Hash` as well

Reply:
fixed in [a727237](https://github.com/hashintel/hash/pull/1800/commits/a727237a55d6a1fdcad4abc805e546800833dd14)